src/components/can_cooperation/src/commands/on_interior_vehicle_data_notification.cc

```diff
@@ -0,0 +1,124 @@
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "can_cooperation/commands/on_interior_vehicle_data_notification.h"
+#include <algorithm>
+#include <vector>
+#include "json/json.h"
+#include "utils/make_shared.h"
+#include "can_cooperation/can_module_constants.h"
+#include "can_cooperation/message_helper.h"
+#include "can_cooperation/can_module.h"
+#include "can_cooperation/can_app_extension.h"
+
+namespace can_cooperation {
+
+namespace commands {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "OnInteriorVehicleDataNotification")
+
+OnInteriorVehicleDataNotification::OnInteriorVehicleDataNotification(
+    const application_manager::MessagePtr& message,
+    CANModuleInterface& can_module)
+    : BaseCommandNotification(message, can_module) {}
+
+OnInteriorVehicleDataNotification::~OnInteriorVehicleDataNotification() {}
+
+void OnInteriorVehicleDataNotification::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  Json::Value moduleDescription(Json::ValueType::objectValue);
+  Json::Value json;
+
+  application_manager::MessagePtr msg = message();
+
+  json = MessageHelper::StringToValue(msg->json_message());
+
+  moduleDescription[message_params::kModuleType] =
+      json[message_params::kModuleData][message_params::kModuleType];
+
+  typedef std::vector<application_manager::ApplicationSharedPtr> AppPtrs;
+  AppPtrs apps = service_->GetApplications(can_module_.GetModuleID());
+
+  for (AppPtrs::iterator it = apps.begin(); it != apps.end(); ++it) {
+    DCHECK(*it);
+    application_manager::Application& app = **it;
+
+    CANAppExtensionPtr extension =
+        application_manager::AppExtensionPtr::static_pointer_cast<
+            CANAppExtension>(app.QueryInterface(can_module_.GetModuleID()));
+    DCHECK(extension);
+    if (extension->IsSubscibedToInteriorVehicleData(moduleDescription)) {
+      application_manager::MessagePtr message =
+          utils::MakeShared<application_manager::Message>(*msg);
+      message->set_message_type(
+          application_manager::MessageType::kNotification);
+      message->set_protocol_version(application_manager::kV3);
+      message->set_function_id(functional_modules::ON_INTERIOR_VEHICLE_DATA);
+      message->set_function_name(MessageHelper::GetMobileAPIName(
+          functional_modules::ON_INTERIOR_VEHICLE_DATA));
+      message->set_connection_key(app.app_id());
+      NotifyOneApplication(message);
```

Comment:
@LuxoftAKutsan probably you also should include `moduleData` parameter to this message as it's present in response